ggml.c

```diff
@@ -266,6 +267,24 @@ inline static void* ggml_aligned_malloc(size_t size) {
 #define MIN(a, b) ((a) < (b) ? (a) : (b))
 #define MAX(a, b) ((a) > (b) ? (a) : (b))
 
+#ifdef _MSC_VER
+static float minf(float x, float y) { return x < y ? x : y; }
+static float maxf(float x, float y) { return x > y ? x : y; }
+#else
+// use -ffast-math so these function optimize to vminss and vmaxss
+#ifndef __clang__
+#pragma GCC push_options
+#pragma GCC optimize "-ffast-math"
+#endif
+__attribute__((always_inline))
+static inline float minf(float x, float y) { return x < y ? x : y; }
+__attribute__((always_inline))
+static inline float maxf(float x, float y) { return x > y ? x : y; }
+#ifndef __clang__
+#pragma GCC pop_options
+#endif
+#endif
```

Comment:
Don't think we need `minf` and `maxf`. Use the existing `MIN` and `MAX`

Reply:
Yeah, probably premature optimization on my part. I was concerned about adding so many branches to a loop that was previously quite simple, but I haven't done any actual benchmarks.